src/uct/ib/ud/accel/ud_mlx5_common.c

```diff
@@ -15,10 +15,21 @@ ucs_config_field_t uct_ud_mlx5_iface_common_config_table[] = {
   {NULL}
 };
 
-ucs_status_t uct_ud_mlx5_iface_common_init(uct_ud_mlx5_iface_common_t *iface,
+ucs_status_t uct_ud_mlx5_iface_common_init(uct_ib_iface_t *ib_iface,
+                                           uct_ud_mlx5_iface_common_t *iface,
                                            uct_ud_mlx5_iface_common_config_t *config)
 {
+    ucs_status_t status;
+
     iface->config.compact_av = config->enable_compact_av;
+
+    if (iface->config.compact_av) {
```

Comment:
i would check "config->enable_compact_av" in this "if", instead of using it in line 24. and set "iface->config.compact_av" to 0 in the "else"